File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
       int n=prices.size();
        int minprice=prices[0];
        int profit;
        int maxprofit=0;
        for(int i=0;i<n;i++){
            if(prices[i]<minprice){
                minprice=prices[i];
            }
            else{
                profit=prices[i]-minprice;
                maxprofit=max(profit,maxprofit);
            }
        }
        return maxprofit;  
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
       n=len(prices)
       min_price=float("inf")
       profit=0
       for i in range(0,n):

        min_price=min(min_price,prices[i])
        profit=max(profit,prices[i]-min_price)
       return profit
       